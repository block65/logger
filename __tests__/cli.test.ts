import { beforeEach, describe, expect, jest, test } from '@jest/globals';

describe('CLI', () => {
  const oldEnv = process.env;
  beforeEach(() => {
    jest.resetModules();
    process.env = { ...oldEnv };
  });

  test('pretty logger no-color', async () => {
    process.env.NO_COLOR = 'true';
    const { cliLoggerWithWaitableMockWatchOnce } = await import('./helpers.js');

    const [logger, callback] = cliLoggerWithWaitableMockWatchOnce();
    logger.error(new Error('hallo'));
    logger.info(new Error('hello'));
    logger.warn(new Error('halo'));
    logger.debug(new Error('gday'));
    logger.trace(new Error('nihao'));
    await expect(callback.waitUntilCalledTimes(1)).resolves.toMatchSnapshot();
  });

  test('pretty logger force-color', async () => {
    const { cliLoggerWithWaitableMockWatchOnce } = await import('./helpers.js');

    const [logger, callback] = cliLoggerWithWaitableMockWatchOnce({
      color: true,
    });
    logger.error(new Error('hallo'));
    logger.info(new Error('hello'));
    logger.warn(new Error('halo'));
    logger.debug(new Error('gday'));
    logger.trace(new Error('nihao'));
    await expect(callback.waitUntilCalledTimes(1)).resolves.toMatchSnapshot();
  });
});
