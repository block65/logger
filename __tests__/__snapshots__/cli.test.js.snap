// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI force-color off 1`] = `
Array [
  Array [
    "2009-02-13T23:31:30.000Z FATAL  herp { err: { message: 'herp', name: 'Error', stack: 'Error: herp\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:36:22)\\\\n    at runMicrotasks (<anonymous>)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)' } }
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z ERROR  hallo { err: { message: 'hallo', name: 'Error', stack: 'Error: hallo\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:37:22)\\\\n    at runMicrotasks (<anonymous>)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)' } }
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z INFO   hello { err: { message: 'hello', name: 'Error', stack: 'Error: hello\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:38:21)\\\\n    at runMicrotasks (<anonymous>)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)' } }
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z WARN   halo { err: { message: 'halo', name: 'Error', stack: 'Error: halo\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:39:21)\\\\n    at runMicrotasks (<anonymous>)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)' } }
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z DEBUG  gday { err: { message: 'gday', name: 'Error', stack: 'Error: gday\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:40:22)\\\\n    at runMicrotasks (<anonymous>)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)' } }
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z TRACE  nihao3 { err: { message: 'nihao3', name: 'Error', stack: 'Error: nihao3\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:41:22)\\\\n    at runMicrotasks (<anonymous>)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)' } }
",
  ],
]
`;

exports[`CLI force-color on 1`] = `
Array [
  Array [
    "[90m2009-02-13T23:31:30.000Z[39m [41m[97m[1mFATAL[22m[39m[49m [1mherp[22m { err: { message: [32m'herp'[39m, name: [32m'Error'[39m, stack: [32m'Error: herp\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:18:22)'[39m } }
",
  ],
  Array [
    "[90m2009-02-13T23:31:30.000Z[39m [31mERROR[39m [1mhallo[22m { err: { message: [32m'hallo'[39m, name: [32m'Error'[39m, stack: [32m'Error: hallo\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:19:22)'[39m } }
",
  ],
  Array [
    "[90m2009-02-13T23:31:30.000Z[39m [34mINFO[39m [1mhello[22m { err: { message: [32m'hello'[39m, name: [32m'Error'[39m, stack: [32m'Error: hello\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:20:21)'[39m } }
",
  ],
  Array [
    "[90m2009-02-13T23:31:30.000Z[39m [43m[30m[1mWARN[22m[39m[49m [1mhalo[22m { err: { message: [32m'halo'[39m, name: [32m'Error'[39m, stack: [32m'Error: halo\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:21:21)'[39m } }
",
  ],
  Array [
    "[90m2009-02-13T23:31:30.000Z[39m [32mDEBUG[39m [1mgday[22m { err: { message: [32m'gday'[39m, name: [32m'Error'[39m, stack: [32m'Error: gday\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:22:22)'[39m } }
",
  ],
  Array [
    "[90m2009-02-13T23:31:30.000Z[39m [2mTRACE[22m [1mnihao3[22m { err: { message: [32m'nihao3'[39m, name: [32m'Error'[39m, stack: [32m'Error: nihao3\\\\n    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/cli.test.js:23:22)'[39m } }
",
  ],
]
`;
